go/client/cmd_pgp_gen.go

```diff
@@ -68,6 +73,32 @@ func (v *CmdPGPGen) Run() (err error) {
 	return err
 }
 
+// Duplicates the test in libkb.PGPGenArg.CreatePGPIDs() for interactive input
+var CheckPGPID = libkb.Checker{
+	F: func(s string) bool {
+		if !strings.Contains(s, "<") && libkb.CheckEmail.F(s) {
+			return true
+		}
+		_, err := libkb.ParseIdentity(s)
+		return err == nil
+	},
+	Hint: "2-12 letter id or email, or pgp style id",
+}
+
+func (v *CmdPGPGen) propmptPGPIDs() (err error) {
+	prompt := "Enter default ID"
```

Comment:
Ok great so a multi step input process:

1. Explain what is going on
2. Enter the full name you want to appear in your pgp key
3. Enter your email address
4. Enter you second email address
5. Etc

With sensible error checking at each step.  And control C should always
fully cancel the experience.

Does that sound right?

Of course the issue with the current code is that no one will know what
"ID" means in this case.

On Tuesday, January 12, 2016, Chris Coyne <notifications@github.com> wrote:

> In go/client/cmd_pgp_gen.go
> <https://github.com/keybase/client/pull/1706#discussion_r49454966>:
>
> > @@ -68,6 +73,32 @@ func (v *CmdPGPGen) Run() (err error) {
> >  	return err
> >  }
> >
> > +// Duplicates the test in libkb.PGPGenArg.CreatePGPIDs() for interactive input
> > +var CheckPGPID = libkb.Checker{
> > +	F: func(s string) bool {
> > +		if !strings.Contains(s, "<") && libkb.CheckEmail.F(s) {
> > +			return true
> > +		}
> > +		_, err := libkb.ParseIdentity(s)
> > +		return err == nil
> > +	},
> > +	Hint: "2-12 letter id or email, or pgp style id",
> > +}
> > +
> > +func (v *CmdPGPGen) propmptPGPIDs() (err error) {
> > +	prompt := "Enter default ID"
>
> yeah, when people do this in interactive mode, we're putting them on
> rails, so to speak, not giving them all the possibilities of PGP. they
> should be entering a full name once, a regular email address, etc.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/keybase/client/pull/1706/files#r49454966>.
>